ingestion/create_cadet_databases_source/source.py

```diff
@@ -279,11 +289,22 @@ def _get_databases_with_domains_and_display_tags(
                         logging.debug(f"{database} - has no database level metadata")
 
                     database_metadata_dict["domain"] = fqn[1]
-                    database_metadata_tuple = tuple(database_metadata_dict.items())
+                    database_tags = database_metadata_dict.get("tags", [])
+                    # A copy is needed to avoid changing the original dict, whish is reused
+                    database_metadata_dict_copy = database_metadata_dict.copy()
+                    # Tags are a list which is unhashable for the tuple so it needs to be removed
```

Comment:
I think it would be clearer to explicitly create the tuple with the items you do want here, rather than dynamically creating it from the `.items()` value - this way you don't need the special logic to remove the tags entry and anyone reading this code can see how many things are in the tuple.

I also noticed the type signature is wrong (unrelated to this PR) and the docstring is also a little out of date, would you mind updating those as well?

Overall this method is doing quite a lot now due to it populating 3 different data structures, so I suspect there is more we could do to simplify it (e.g. by changing the return value and/or extracting helpers). I'm not suggesting we do a big refactor now, but maybe it's worth looking at it the next time we need to make a change here.